### ISSUE TITLE
Training resnet34 (from scratch) on CIFAR-10 leads to nan loss 

### ISSUE THREAD

**Issue by stack-dhruv** [xuyuankun631/IcicleGCN#3]
I am training the resnet34 from scratch on CIFAR-10 dataset, also changed some params in train.py as well. 

```yaml
# --config.yaml--

# general
seed: 42
workers: 4
dataset_dir: "./datasets"

# train options
batch_size: 100
image_size: 224
start_epoch: 0
epochs: 1015
dataset: "CIFAR-10" #  CIFAR-10 / CIFAR-100 / STL-10 / ImageNet-10 / ImageNet-dogs / tiny-ImageNet / Flowers17

# model options
resnet: "ResNet34" # ResNet18 / ResNet34 / ResNet50
feature_dim: 128
model_path: "./save/CIFAR-10"
reload: False

# loss options
learning_rate: 0.001 #
weight_decay: 0.0001
instance_temperature: 0.5
cluster_temperature: 1.0
```

Training the model from scratch lead to nan just in a single epoch - 
```sh
# --terminal output--
Files already downloaded and verified
Files already downloaded and verified
Step [0/600]       loss_instance: 5.28515625      loss_cluster: 2.946188449859619        re_loss: 1.126003384590149      
Step [50/600]      loss_instance: 5.009121417999268       loss_cluster: 2.4746360778808594       re_loss: 0.034027669578790665   
Step [100/600]     loss_instance: 4.926772117614746       loss_cluster: 2.2181878089904785       re_loss: 0.024739207699894905   
Step [150/600]     loss_instance: 4.885540962219238       loss_cluster: 2.2527127265930176       re_loss: 0.04029698297381401    
Step [200/600]     loss_instance: 4.774231910705566       loss_cluster: 2.339139699935913        re_loss: 0.05486036837100983    
Step [250/600]     loss_instance: 4.869399547576904       loss_cluster: 2.6640102863311768       re_loss: 0.03202003985643387    
Step [300/600]     loss_instance: 5.103700160980225       loss_cluster: 2.4775240421295166       re_loss: 0.034477513283491135   
Step [350/600]     loss_instance: 4.827446937561035       loss_cluster: 2.7474653720855713       re_loss: 0.03616001829504967    
Step [400/600]     loss_instance: 4.9964423179626465      loss_cluster: 2.49931263923645         re_loss: 0.03911386430263519    
Step [450/600]     loss_instance: 5.18294620513916        loss_cluster: 2.3889520168304443       re_loss: 0.029943885281682014   
Step [500/600]     loss_instance: nan     loss_cluster: nan      re_loss: nan    
Step [550/600]     loss_instance: nan     loss_cluster: nan      re_loss: nan 
```

I tried default training params as well, but it also led to nan after around 60 epochs. Is there anything I am missing over here?

**Comment by xuyuankun631** [comment]
start_epoch should be set to 1000. Due to git reasons, the original checkpoint_1000.tar weight file was not uploaded because it was too large. You can get the 1000-round weight files of cifar-10 and cifar-100 in the following repository:: https://github.com/Yunfan-Li/Contrastive-Clustering/tree/main/save.

**Comment by stack-dhruv** [comment]
What if I want to train the model from scratch and on the custom dataset?

**Comment by xuyuankun631** [comment]
You can use the train.py and network.py in the following article for pre-training(https://github.com/Yunfan-Li/Contrastive-Clustering/blob/main/train.py) and the weights obtained can be used as the worm-up of IcicleGCN. The network of IcicleGCN is slightly different from that in the previous article. You can think of IcicleGCN as a boosting algorithm on the CC model.

**Comment by stack-dhruv** [comment]
> You can use the train.py and network.py in the following article for pre-training(https://github.com/Yunfan-Li/Contrastive-Clustering/blob/main/train.py) and the weights obtained can be used as the worm-up of IcicleGCN. The network of IcicleGCN is slightly different from that in the previous article. You can think of IcicleGCN as a boosting algorithm on the CC model.

Thanks
